src/shared/networkmanager.cpp

```diff
@@ -15,13 +17,30 @@
 
 namespace {
 NetworkManager* s_instance = nullptr;
+
+bool localhostRequestCallback(NetworkRequest* request) {
+  QString host(request->url().host());
+  if (host == "localhost" || host == "127.0.0.1" || host == "::1") {
```

Comment:
Praise: I saw the tests failing here before in the CI. Nice workaround with the captive portal! :)